contracts/standard/arbitration/composed-arbitrable/example/ArbitrableTokenList.sol

```diff
@@ -0,0 +1,594 @@
+     *  @return The numbers of items in the list per status.
+     */
+    function itemsCounts() public view returns (uint pending, uint challenged, uint accepted, uint rejected) {
+        for (uint i = 0; i < itemsList.length; i++) {
+            Item storage item = items[itemsList[i]];
+            Agreement storage latestAgreement = agreements[latestAgreementId(itemsList[i])];
+            if (latestAgreement.disputed) challenged++;
+            else if (item.status == ItemStatus.Resubmitted || item.status == ItemStatus.Submitted) pending++;
+            else if (item.status == ItemStatus.Registered) accepted++;
+            else if (item.status == ItemStatus.Cleared) rejected++;
+        }
+    }
+
+    /**
+     *  @dev Return the values of the items the query finds.
+     *  This function is O(n) at worst, where n is the number of items. This could exceed the gas limit, therefore this function should only be used for interface display and not by other contracts.
+     *  @param _cursor The pagination cursor.
+     *  @param _count The number of items to return.
+     *  @param _filter The filter to use.
+     *  @param _sort The sort order to use.
+     *  @return The values of the items found and wether there are more items for the current filter and sort.
+     */
+    function queryItems(bytes32 _cursor, uint _count, bool[6] _filter, bool _sort) public view returns (bytes32[] values, bool hasMore) {
+        uint _cursorIndex;
+        values = new bytes32[](_count);
+        uint _index = 0;
+
+        if (_cursor == 0)
+            _cursorIndex = 0;
+        else {
+            for (uint j = 0; j < itemsList.length; j++) {
+                if (itemsList[j] == _cursor) {
+                    _cursorIndex = j;
+                    break;
+                }
+            }
+            require(_cursorIndex != 0, "The cursor is invalid.");
+        }
+
+        for (
+                uint i = _cursorIndex == 0 ? (_sort ? 0 : 1) : (_sort ? _cursorIndex + 1 : itemsList.length - _cursorIndex + 1);
+                _sort ? i < itemsList.length : i <= itemsList.length;
+                i++
+            ) { // Oldest or newest first.
+            bytes32 itemID = itemsList[_sort ? i : itemsList.length - i];
+            Item storage item = items[itemID];
+            bytes32 agreementId = latestAgreementId(itemID);
+            Agreement storage agreement = agreements[agreementId];
+            if (
+                    // solium-disable-next-line operator-whitespace
+                    item.status != ItemStatus.Absent && item.status != ItemStatus.PreventiveClearingRequested && (
+                    // solium-disable-next-line operator-whitespace
+                    (_filter[0] && (item.status == ItemStatus.Resubmitted || item.status == ItemStatus.Submitted)) || // Pending.
+                    (_filter[1] && agreement.disputed) || // Challenged.
+                    (_filter[2] && item.status == ItemStatus.Registered) || // Accepted.
```

Comment:
Let's keep My Challenges and My Submissions, and map the rest to the actual statuses in the enum, just like with itemsCount.